File: courtlistener/api/debts.py
"""
Debts API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class DebtsAPI(BaseAPI):
    """API for accessing debt and liability data from financial disclosures."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "debts/"
    
    def list(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List debt records.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of debts
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, debt_id: int) -> Dict[str, Any]:
        """
        Get a specific debt by ID.
        
        Args:
            debt_id: Debt ID
        
        Returns:
            Debt data
        """
        return self.client.get(f"{self.endpoint}{debt_id}/")
    
    def paginate(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated debts.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for debts
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/Debts.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Debts API endpoint
 */
class Debts extends BaseApi
{
    protected string $endpoint = 'debts/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List debts with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listDebts(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific debts by ID
     *
     * @param int|string $id Debts ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getDebts(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search debts
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchDebts(array $params = []) {
        return $this->search($params);
    }
}
